web/src/components/tasks/business-formation/NexusFormation.test.tsx

```diff
@@ -310,6 +311,106 @@ describe("<NexusFormationFlow />", () => {
+          });
+          const formationData = generateFormationData({ lastVisitedPageIndex });
+          displayContent = {
+            formationDbaContent: generateFormationDbaContent({}),
+          };
+          return (initialBusiness = generateBusiness({ profileData, formationData }));
+        };
+
+        beforeEach(() => {
+          jest.resetAllMocks();
+          useSetupInitialMocks();
+        });
+
+        it("shows the nexus name step with an initial user when lastVisitedPage has index 0 which is by default", async () => {
+          preparePage({ business: getNexusBusiness({ lastVisitedPageIndex: 0 }), displayContent });
+          expect(screen.getByTestId("nexus-name-step")).toBeInTheDocument();
+        });
+
+        it("shows the last formation step when lastVisitedPage has step index 4", () => {
+          preparePage({
+            business: getNexusBusiness({ lastVisitedPageIndex: 4, businessName: "fake-business-name" }),
+            displayContent,
+          });
+          expect(screen.getByTestId("review-step")).toBeInTheDocument();
+        });
+
+        it("lastVisitedPage updates on step change via stepper", async () => {
+          page = preparePage({
+            business: getNexusBusiness({ lastVisitedPageIndex: 4, businessName: "fake-business-name" }),
+            displayContent,
+          });
+          expect(screen.getByTestId("review-step")).toBeInTheDocument();
+
+          await page.stepperClickToBillingStep();
+          await waitFor(() => {
+            expect(currentBusiness().formationData.lastVisitedPageIndex).toEqual(3);
+          });
+
+          await page.stepperClickToReviewStep();
+          await waitFor(() => {
+            expect(currentBusiness().formationData.lastVisitedPageIndex).toEqual(4);
+          });
+        });
+
+        it("lastVisitedPage updates on step change via the previous and next button steps", async () => {
+          page = preparePage({
+            business: getNexusBusiness({ lastVisitedPageIndex: 4, businessName: "fake-business-name" }),
+            displayContent,
+          });
+
+          await waitFor(() => {
+            expect(currentBusiness().formationData.lastVisitedPageIndex).toEqual(4);
+          });
+          expect(screen.getByTestId("review-step")).toBeInTheDocument();
+
+          fireEvent.click(screen.getByTestId("previous-button"));
+          await waitFor(() => {
+            expect(currentBusiness().formationData.lastVisitedPageIndex).toEqual(3);
+          });
+          expect(screen.getByTestId("billing-step")).toBeInTheDocument();
+
+          fireEvent.click(screen.getByTestId("next-button"));
+          await waitFor(() => {
+            expect(currentBusiness().formationData.lastVisitedPageIndex).toEqual(4);
+          });
+          expect(screen.getByTestId("review-step")).toBeInTheDocument();
+        });
+
+        it("defaults to first formation step we go beyond page index allowed", () => {
```

Comment:
[pebble] We should update the title to something like `"defaults to first formation step when we go beyond page index of 5"` or `"defaults to first formation step when we go beyond 5 for page indexes"`. The latter would follow the way we're naming the next test.

Reply:
good call! updated it to `defaults to first formation step when we go above 4 for page indexes` as 0 to 4 inclusive are the valid step indices